### PR TITLE
Fix issue #24635: Use of undefined constant GLOB_BRACE in dependency …

### LINKED ISSUE #24635: Use of undefined constant GLOB_BRACE in dependency "allure-framework/allure-phpunit"
### Preconditions (*)
Magento Version: 2.3.2

### Steps to reproduce (*)

`./bin/magento dev:test:run unit`

### Expected result (*)
No PHP Warnings

### Actual result (*)

several:
```
Warning: Use of undefined constant GLOB_BRACE - assumed 'GLOB_BRACE' (this will throw an Error in a future version of PHP) in /var/www/html/src/vendor/allure-framework/allure-phpunit/src/Yandex/Allure/Adapter/AllureAdapter.php:74.
```

### Related Informations

Problem is, GLOB_BRACE is not available in alle Unix Systems.
For example the often in Docker context used "Alpine" 
https://github.com/docker-library/php/issues/719

There is an (not merged) upstream PullRequest available in https://github.com/allure-framework/allure-phpunit/pull/44
But I did not check yet, if this is the proper solution.

### PR DESCRIPTION
**Pull request by vidyli**
### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/441
https://github.com/magento-commerce/magento2-infrastructure/pull/1516

Fixes #24635

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@vidyli could you add to the PR description the following line?
> Fixes https://github.com/magento/magento2/issues/24635

@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Static Tests, Unit Tests

**Comment by ihor-sviziev** [comment]
@vidyli Seems like some changes should be done also on EE or/and B2B repositories.
I see you don't have access to them, so we'll ask someone from the internal team to check it

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@engcom-Golf the tests still failing

**Comment by engcom-Golf** [comment]
I see. Requires changes to infrastructure repo. Will address it. On Dec 16, 2020 6:21 PM, Ihor Sviziev <notifications@github.com> wrote:
@engcom-Golf the tests still failing

**Comment by engcom-Golf** [comment]
@magento run all tests

File: composer.lock [hunk 1]
```diff
@@ -6316,7 +6318,13 @@
                 "type": "zip",
                 "url": "https://api.github.com/repos/allure-framework/allure-codeception/zipball/a69800eeef83007ced9502a3349ff72f5fb6b4e2",
                 "reference": "a69800eeef83007ced9502a3349ff72f5fb6b4e2",
-                "shasum": ""
+                "shasum": "",
+                "mirrors": [
```
Comment (ihor-sviziev) [inline composer.lock]: @vidyli,
oh... we shouldn't have mirrors section in the composer.lock files.
Could you configure your composer to not use mirrors and run the `composer update` command?

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@engcom-Golf thx for help!

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, Functional Tests EE, Functional Tests CE, Unit Tests

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@engcom-Golf any thoughts on why unit tests are failing?

File: composer.json [hunk 2]
```diff
@@ -86,7 +86,7 @@
         "league/flysystem-cached-adapter": "^1.0"
     },
     "require-dev": {
-        "allure-framework/allure-phpunit": "~1.2.0",
+        "allure-framework/allure-phpunit": "~1.3.0",
```
Comment (fredden) [inline composer.json]: Version 1.3.0 doesn't contain the fix for the bug you're trying to fix here.
```suggestion
        "allure-framework/allure-phpunit": "^1.3.1",
```

**Review by fredden — CHANGES_REQUESTED** [review]

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests CE, Unit Tests

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Hi @vidyli,
Could you check why the unit tests are failing?

**Review by fredden — APPROVED** [review]

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Unit tests still failing. Please check why and fix that

**Comment by ihor-sviziev** [comment]
@magento run WebAPI Tests, Unit Tests, Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run WebAPI Tests, Unit Tests, Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Hi @engcom-Golf,
Could you help with the `Pull Request not mergeable` issue on tests?
Thank you

**Comment by engcom-Golf** [comment]
@magento run all tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8698](https://jira.corp.magento.com/browse/ENGCOM-8698) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@vidyli thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Review by ihor-sviziev — CHANGES_REQUESTED** [review]
Could you resolve conflicts?

**Comment by gabrieldagama** [comment]
Hi @vidyli, can you update your branch and resolve the conflicts?

Thanks!

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Hi @engcom-Golf,
Could you resolve conflicts in the related PRs as the author doesn't have access to these repositories?

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
Hi @engcom-Foxtrot,
Could you update this dependency to at least 1.4.0 https://packagist.org/packages/allure-framework/allure-phpunit#1.4.0, as this is a requirement for updating another extension in https://github.com/magento/magento2/pull/32614 ?

**Comment by ihor-sviziev** [comment]
@engcom-Foxtrot, tests still failing :(

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by engcom-Foxtrot** [comment]
@magento run all tests

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Comment by ihor-sviziev** [comment]
@magento run Integration Tests

**Review by ihor-sviziev — APPROVED** [review]

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8698](https://jira.corp.magento.com/browse/ENGCOM-8698) has been created to process this Pull Request

**Comment by ihor-sviziev** [comment]
@engcom-oscar, could you take this issue in testing, as it's blocking the https://github.com/magento/magento2/pull/32614?

**Comment by ihor-sviziev** [comment]
I added label project php8, as we have dependency on this PR there.

**Comment by sivaschenko** [comment]
@ihor-sviziev removed "Project: PHP8" label to merge this PR together with other PRs from high priority board

**Pull request merged**